code/game/machinery/dance_machine.dm

```diff
@@ -342,128 +340,69 @@
-				initial_matrix.Translate(0,3)
-				animate(M, transform = initial_matrix, time = 1, loop = 0)
-			if (SOUTH)
-				initial_matrix = matrix(M.transform)
-				initial_matrix.Translate(0,-3)
-				animate(M, transform = initial_matrix, time = 1, loop = 0)
-			if (EAST)
-				initial_matrix = matrix(M.transform)
-				initial_matrix.Translate(3,0)
-				animate(M, transform = initial_matrix, time = 1, loop = 0)
-			if (WEST)
-				initial_matrix = matrix(M.transform)
-				initial_matrix.Translate(-3,0)
-				animate(M, transform = initial_matrix, time = 1, loop = 0)
-		sleep(0.1 SECONDS)
-	M.lying_fix()
-
-/obj/machinery/jukebox/disco/proc/dance4(mob/living/lead_dancer)
-	var/speed = rand(1,3)
-	set waitfor = 0
-	var/time = 30
-	while(time)
-		sleep(speed)
-		for(var/i in 1 to speed)
-			lead_dancer.setDir(pick(GLOB.cardinals))
-			// makes people dance with us nearby
-			for(var/datum/weakref/weak_dancer as anything in rangers)
-				var/mob/living/carbon/dancer = weak_dancer.resolve()
-				if(!istype(dancer))
-					continue
-				dancer.set_resting(!dancer.resting, silent = TRUE, instant = TRUE)
-		time--
-
-/obj/machinery/jukebox/disco/proc/dance5(mob/living/M)
-	animate(M, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
-	var/matrix/initial_matrix = matrix(M.transform)
-	for (var/i in 1 to 60)
-		if (!M)
-			return
-		if (i<31)
-			initial_matrix = matrix(M.transform)
-			initial_matrix.Translate(0,1)
-			animate(M, transform = initial_matrix, time = 1, loop = 0)
-		if (i>30)
-			initial_matrix = matrix(M.transform)
-			initial_matrix.Translate(0,-1)
-			animate(M, transform = initial_matrix, time = 1, loop = 0)
-		M.setDir(turn(M.dir, 90))
-		switch (M.dir)
-			if (NORTH)
-				initial_matrix = matrix(M.transform)
-				initial_matrix.Translate(0,3)
-				animate(M, transform = initial_matrix, time = 1, loop = 0)
-			if (SOUTH)
-				initial_matrix = matrix(M.transform)
-				initial_matrix.Translate(0,-3)
-				animate(M, transform = initial_matrix, time = 1, loop = 0)
-			if (EAST)
-				initial_matrix = matrix(M.transform)
-				initial_matrix.Translate(3,0)
-				animate(M, transform = initial_matrix, time = 1, loop = 0)
-			if (WEST)
-				initial_matrix = matrix(M.transform)
-				initial_matrix.Translate(-3,0)
-				animate(M, transform = initial_matrix, time = 1, loop = 0)
-		sleep(1 SECONDS)
-	M.lying_fix()
-
-/mob/living/proc/lying_fix()
-	animate(src, transform = null, time = 1, loop = 0)
-	lying_prev = 0
+/obj/machinery/jukebox/disco/proc/dance1(mob/living/dancer)
+	addtimer(TRAIT_CALLBACK_REMOVE(dancer, TRAIT_DISCO_DANCER, REF(src)), 6.5 SECONDS, TIMER_CLIENT_TIME)
+	for(var/i in 0 to (6 SECONDS) step (1.5 SECONDS))
+		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(dance_rotate), dancer, CALLBACK(dancer, TYPE_PROC_REF(/mob, dance_flip))), i, TIMER_CLIENT_TIME)
+
+/obj/machinery/jukebox/disco/proc/dance2(mob/living/dancer, dance_length = 2.5 SECONDS)
+	var/matrix/initial_matrix = matrix(dancer.transform)
```

Comment:
Resetting the matrix to what it was seconds ago may cause issues if something happens to modify the mob's transform matrix between the start and the end of the dance.
Instead of a reset, inverse animations should be performed to revert the ongoing visuals (eg `Turn(-90)` if `Turn(90)` was previously called).

Yes, that's how it was before, but it's also part of the clean up too, so pretty please and thank you.

Reply:
it may but this is an end of round thing

Reply:
and i dont think the effort is worth it

Reply:
You don't think dance machines can be spawned outside of the escape shuttle?

Reply:
admins can also gib and crash people and massedit transforms it is not my fault